storage/innobase/xtrabackup/test/t/innodb_keyring_file_component.sh

```diff
@@ -0,0 +1,61 @@
+#
+# Component keyring file specific variables
+#
+require_server_version_higher_than 8.0.23
+
+KEYRING_TYPE="component"
+. inc/keyring_common.sh
+. inc/keyring_file.sh
+configure_server_with_component
+
+run_cmd $MYSQL $MYSQL_ARGS test <<EOF
+CREATE TABLE t1 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t1 (c1) VALUES ('ONE'), ('TWO'), ('THREE');
+INSERT INTO t1 (c1) VALUES ('10'), ('20'), ('30');
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+INSERT INTO t1 SELECT * FROM t1;
+ALTER INSTANCE ROTATE INNODB MASTER KEY;
+CREATE TABLE t2 (c1 VARCHAR(100)) ENCRYPTION='y';
+INSERT INTO t2 SELECT * FROM t1;
+EOF
+
+xtrabackup --backup --target-dir=$topdir/backup1
+cp -r $topdir/backup1 $topdir/backup2
+record_db_state test
+shutdown_server
+
+xtrabackup --stats --datadir=${mysql_datadir} \
+  --component-keyring-file-config=${MYSQLD_DATADIR}/component_keyring_file.cnf \
+  --xtrabackup-plugin-dir=${plugin_dir}
+
+if grep -q "Encryption can't find master key, please check the keyring is loaded." $OUTFILE
```

Comment:
OUTFILE is set in common header?

Reply:
Yes, adjusted this a bit. I always miss a function to pass a pattern, file and occurrences I expect and that will exit in case the number of occurrences of patter on file doesn't match. Adjusted xtrabackup commands inside this test case to tee the output to a specific file and used that new file (which gets overwritten every time) as the source of grep file.